cli/flags.rs

```diff
@@ -259,6 +262,8 @@ pub fn flags_from_vec_safe(args: Vec<String>) -> clap::Result<Flags> {
     upgrade_parse(&mut flags, m);
   } else if let Some(m) = matches.subcommand_matches("doc") {
     doc_parse(&mut flags, m);
+  } else if let Some(m) = matches.subcommand_matches("lint") {
+    lint_parse(&mut flags, m);
```

Comment:
I don't know Rust, but is there an early return in Rust? If there is, wouldn't it be better if the file was something like:
```rust
...
  if let Some(m) = matches.subcommand_matches("doc") {
    return doc_parse(&mut flags, m);
  }
  if let Some(m) = matches.subcommand_matches("lint") {
    return lint_parse(&mut flags, m);
  }
```

Reply:
This method returns `Result<Flags>` while each of the methods returns `()`.

Reply:
Got it. My focus was not on the returning part but on possibility of eliminating this many `if-else`s :) Perhaps a switch statement? 
Again, this is totally trivial and unrelated. I was just looking at that file generally. Awesome work btw.